sdk/src/main/java/com/mapbox/maps/renderer/MapboxRenderThread.kt

```diff
@@ -242,17 +240,14 @@ internal class MapboxRenderThread : Choreographer.FrameCallback {
     }
   }
 
-  private fun draw() {
-    val renderTimeNsCopy = renderTimeNs
-    val currentTimeNs = SystemClock.elapsedRealtimeNanos()
-    val expectedEndRenderTimeNs = currentTimeNs + renderTimeNsCopy
-    if (expectedVsyncWakeTimeNs > currentTimeNs) {
+  private fun draw(frameTimeNanos: Long) {
+    val proceedFrameRender = fpsManager.preRender(frameTimeNanos)
+    if (!proceedFrameRender) {
```

Comment:
Just `if (!fpsManager.preRender(frameTimeNanos)) {` ?

Reply:
why not